### ISSUE TITLE
off-chain worker tutorials and explanation needs to be clear and have more explanations

### ISSUE THREAD

**Issue by MrishoLukamba** [substrate-developer-hub/substrate-docs#1064]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Content request

First in the offchain workers tutorials , there are alot of traits used without even a brief explanation untill someone goes to rust docs
 Secondly, some line of codes dont make sense and the explanation is very vague. For example when creating a signed transaction, 
Thirdly there are some crates which share same modules and can be quite confusing when importing them, for example the KeyTypeId module

Fourthly, Please make even a latest tutorial on offchain workers,even in substrate seminars

### Are you willing to help with this request?

Yes!

**Comment by rvalle** [comment]
I agree with @MrishoLukamba the off-chain tutorial is not an actual tutorial, but rather a description, for someone already experience on how to add an off-chain worker to a pallet.

This tutorial needs to be re-worked so that it follows the same pattern as the others, can be followed step by step, starting from the node template, or a previous tutorial, and ends in a working state.

It could, for example, build on top of the "Use macros in a custom pallet" tutorial to add an off chain worker to it.

**Comment by MrishoLukamba** [comment]
> I agree with @MrishoLukamba the off-chain tutorial is not an actual tutorial, but rather a description, for someone already experience on how to add an off-chain worker to a pallet.
> 
> This tutorial needs to be re-worked so that it follows the same pattern as the others, can be followed step by step, starting from the node template, or a previous tutorial, and ends in a working state.
> 
> It could, for example, build on top of the "Use macros in a custom pallet" tutorial to add an off chain worker to it.

I will be happy to assist you and hopefully prepare a tutorial and its true until now Offchain workers have less explanations for a person who is starting

**Comment by rvalle** [comment]
@MrishoLukamba I actually need this tutorial myself. I am not experienced, but I am certainly happy to participate in reworking this tutorial as part of my learning curve. 

I think we could use the "Use macros in a custom pallet" as base. This tutorials allow for registering file hashes on chain as some kind of proof in time.

Perhaps an OCW could disallow hashes already known off chain to be claimable. For an example, perhaps files in a particular directory could represent the outside world. 

I guess this needs to be implemented in the simplest possible way.  perhaps the OCW can be used to claim the files in a particular directory itself. 

@MrishoLukamba  @lisa-parity what do you think?

**Comment by MrishoLukamba** [comment]
Didnt quiet understand what do you want to achieve with OCW there. We can do that tutorial step by step and go over OCW and what they are and their behaviour

**Comment by rvalle** [comment]
Perhaps we start by "Use macros in a custom pallet"  and implement a "Hello World" OCW as a first step. 

I was thinking on giving the OCW some kind of meaningful but minimal functionality for the example in context.

**Comment by MrishoLukamba** [comment]
> Perhaps we start by "Use macros in a custom pallet" and implement a "Hello World" OCW as a first step.
> 
> I was thinking on giving the OCW some kind of meaningful but minimal functionality for the example in context.

Okey sure

**Comment by sacha-l** [comment]
Hey @MrishoLukamba and @rvalle -  I was going to open a new issue to flag the fact that we really should update the OCW learning material in general as they present a lot of misconceptions about how they are used (see [this write-up](https://forum.polkadot.network/t/offchain-workers-design-assumptions-vulnerabilities/2548) about the vulnerabilities of using them).

But seeing that the discussion is already happening here, please allow me to simply rename the issue to "Update and clarify off-chain worker learning material". And @rvalle if you do feel like jumping in on this, please make sure to read what @bernardoaraujor wrote in the forum to inform any new content to write or old content to remove.
